portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,19 +37,23 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
     List<Comment> comments = new ArrayList<>();
     // Iterate over all comments in datastore and add them to array
     for (Entity entity : results.asIterable()) {
```

Comment:
This is kind of wasteful, you're fetching all the rows.
The limiting should be at the lowest level possible (when the query is being executed). 

Hint: use the PreparedQuery#asIterable method with the FetchOptions argument.